### PR TITLE
Change the package manager to go modules

### PR DESCRIPTION
**Pull request by zier-one**
### What problem does this PR solve? 

Change the package manager to go modules

### What is changed and how it works?

we can use go mod to manage dependencies,old 'make xxx' command still work.
we can delete the 'vendor' dir,go mod and makefile will help us handle it

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
why remove vendor? I think we can keep it

**Comment by zier-one** [comment]
> why remove vendor? I think we can keep it

i think it is redundant.
when we use 'make build' or 'make test',it can help us download dependencies and copy them to vendor.we can use vendor as usual,but i think it‘s not necessary to commit vendor in git repo,vendor shouldn't be managed by go.mod and git repo, however only be managed by go.mod.

**Comment by IANTHEREAL** [comment]
vendor is cache here. consider network downloads in china

**Comment by zier-one** [comment]
> vendor is cache here. consider network downloads in china

ok,this is a reason that is difficult to refuse

**Comment by kennytm** [comment]
Must we use TiDB 2.0.8 or could we use the master version to get rid of the yacc parser?

**Comment by IANTHEREAL** [comment]
I think we should use master version in master branch. This repo would release with TiDB after 2.1 ga consistently

**Comment by IANTHEREAL** [comment]
LGTM

File: vendor/github.com/BurntSushi/toml/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+TAGS
```
Comment (WangXiangUSTC) [inline vendor/github.com/BurntSushi/toml/.gitignore]: we can run a script to clean these file, you can see hack/clean_vendor.sh in tidb

Comment (zier-one) [reply vendor/github.com/BurntSushi/toml/.gitignore]: already fix it

**Comment by WangXiangUSTC** [comment]
LGTM

**Comment by tiancaiamao** [comment]
Could `go get` work with this repo now?

**Pull request merged**